src/snapred/backend/recipe/FetchGroceriesRecipe.py

```diff
@@ -17,52 +18,113 @@
+        if item.useLiteMode:
             runNameBuilder.lite(wng.Lite.TRUE)
         return runNameBuilder
 
-    def _createNexusWorkspaceName(self, runConfig: RunConfig) -> str:
-        return self._createNexusWorkspaceNameBuilder(runConfig).build()
+    def _createNexusWorkspaceName(self, item: GroceryListItem) -> str:
+        return self._createNexusWorkspaceNameBuilder(item).build()
+
+    def _createRawNexusWorkspaceName(self, item: GroceryListItem) -> str:
+        return self._createNexusWorkspaceNameBuilder(item).auxilary("Raw").build()
+
+    def _createCopyNexusWorkspaceName(self, item: GroceryListItem, numCopies: int) -> str:
+        return self._createNexusWorkspaceNameBuilder(item).auxilary(f"Copy{numCopies}").build()
 
-    def _createRawNexusWorkspaceName(self, runConfig: RunConfig) -> str:
-        return self._createNexusWorkspaceNameBuilder(runConfig).auxilary("Raw").build()
+    def _createGroupingWorkspaceName(self, item: GroceryListItem):
+        if item.groupingScheme == "Lite":
+            return "lite_grouping_map"
+        instr = "lite" if item.useLiteMode else "native"
+        return Config["grouping.workspacename." + instr] + item.groupingScheme
 
-    def _createCopyNexusWorkspaceName(self, runConfig: RunConfig, numCopies: int) -> str:
-        return self._createNexusWorkspaceNameBuilder(runConfig).auxilary(f"Copy{numCopies}").build()
+    ## FETCH METHODS
+
+    def executeRecipe(self, groceries: List[GroceryListItem]) -> Dict[str, Any]:
+        """
+        inputs:
+        - groceries -- a list of GroceryListItems indicating the workspaces to create
+        outputs a dictionary with keys:
+        - "result": bool, True if everything good, otherwise False
+        - "workspaces": a list of strings workspace names created in the ADS
+        """
+
+        data: Dict[str, Any] = {
+            "result": True,
+            "workspaces": [],
+        }
+        prev: str = ""
+        for item in groceries:
+            self.checkADS(item)
+            if item.workspaceType == "nexus":
+                if item.keepItClean:
+                    res = self.fetchCleanNexusData(item)
+                else:
+                    res = self.fetchDirtyNexusData(item)
+                data["result"] &= res["result"]
+                data["workspaces"].append(res["workspace"])
+                # save the most recently-loaded nexus data as a possible instrument donor for groupings
+                prev = res["workspace"]
+            elif item.workspaceType == "grouping":
+                # if we intend to use the previously loaded workspace as the donor
+                if item.instrumentSource == "prev":
+                    item.instrumentPropertySource = "InstrumentDonor"
+                    item.instrumentSource = prev
+                res = self.fetchGroupingDefinition(item)
+                data["result"] &= res["result"]
+                data["workspaces"].append(res["workspace"])
```

Comment:
if we end up adding another case, may be worth creating a dict of methods keyed on workspace type.

Reply:
The other case we will add will be diffcaltable.  And i think that will be preferable